sdks/java/io/azure/src/test/java/org/apache/beam/sdk/io/azure/blobstore/AzfsResourceIdTest.java

```diff
@@ -0,0 +1,318 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.azure.blobstore;
+
+import static org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions.RESOLVE_DIRECTORY;
+import static org.apache.beam.sdk.io.fs.ResolveOptions.StandardResolveOptions.RESOLVE_FILE;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import java.util.Arrays;
+import java.util.Collection;
+import org.apache.beam.sdk.io.fs.ResolveOptions;
+import org.apache.beam.sdk.io.fs.ResourceId;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.junit.runners.Parameterized;
+
+@RunWith(JUnit4.class)
```

Comment:
It should be `@RunWith(Enclosed.class)` to make sure inner test classes to be picked up.

Reply:
If I switch `@RunWith(JUnit4.class)` to `@RunWith(Enclosed.class)` then all the tests except the inner test class are skipped.

Reply:
I see. I thought you were using subclass with `@RunWith`.  Let's keep with `@RunWith(JUnit4.class)`.

Reply:
@boyuanzz How can I get both the parameterized and the non-parameterized tests to run?  I tried putting all the non-parameterized tests in a separate inner class, but that didn't work either because that class would need to be static for the tests to run but if the class is static then I get an error since `thrown` is non-static.

Reply:
You could something similar to this: https://github.com/apache/beam/blob/d4d45bff2ff010fd317c75b2dcdda61a3843f0da/sdks/java/testing/nexmark/src/test/java/org/apache/beam/sdk/nexmark/queries/sql/SqlQuery0Test.java#L36

Reply:
I just saw your comment now.  You are correct that the inner test classes are being skipped with the current runner, so we shouldn't keep `@RunWith(JUnit4.class)`.

Reply:
If I use `@RunWith(Enclosed.class)` then the non-parameterized tests are skipped unless they are also in an inner class. But if they are in an inner class then the class needs to be static, and that causes an error `non-static variable thrown cannot be referenced from a static context`.

Reply:
Now you should change it to @RunWith(Enclosed.class)